_field-types/supported-field-types/derived.md

```diff
@@ -0,0 +1,903 @@
+
+- Prune the search space by adding query filters on indexed fields together with derived fields.
+- Prefer using doc values over `_source` in the script for faster access, whenever applicable.
+- Consider using a [`prefilter_field`](#prefilter-field) to automatically prune the search space without explicit filters in the search request.
+
+### Prefilter field
+
+Specifying a prefilter field helps prune the search space automatically without adding explicit filters in the search request. The prefilter field specifies an existing indexed field (`prefilter_field`) on which to filter implicitly when constructing the query. The `prefilter_field` must be a text field (either [`text`]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/text/) or [`match_only_text`]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/match-only-text/)).
+
+For example, first update the mapping for the `method` derived field by specifying to prefilter on the `request` field: 
+
+```json
+PUT /logs/_mapping
+{
+  "derived": {
+    "method": {
+      "type": "keyword",
+      "script": {
+        "source": """
+        emit(doc["request.keyword"].value.splitOnToken(" ")[1])
+        """
+      },
+      "prefilter_field": "request"
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Now search using a query on the `method` derived field:
+
+```json
+POST /logs/_search
+{
+  "profile": true,
+  "query": {
+    "term": {
+      "method": {
+        "value": "GET"
+      }
+    }
+  },
+  "fields": ["method"]
+}
+```
+{% include copy-curl.html %}
+
+OpenSearch implicitly adds a filter on the `request` field to your query:
+
+```json
+"#request:GET #DerivedFieldQuery (Query: [ method:GET])"
+```
+
+You can use the `profile` option can to analyze the derived field performance, as shown in the preceding example.
+{: .tip} 
+
+## Dervied object fields
+
+A script can emit a valid JSON object so you can query subfields without indexing them, in the same way as regular fields. This is useful for large JSON objects where occasional searches on some subfields are required. In this case, indexing the subfields is expensive while defining derived fields for each subfield also adds a lot of resource overhead. If you don't [explicitly provide the subfield type](#explicit-subfield-type), the subfield type is [inferred](#inferred-subfield-type).
+
+For example, the following request defines a `derived_request_object` derived field as an `object` type:
+
+```json
+PUT logs_object
+{
+  "mappings": {
+    "properties": {
+      "request_object": { "type": "text" }
+    },
+    "derived": {
+      "derived_request_object": {
+        "type": "object",
+        "script": {
+          "source": "emit(params._source[\"request_object\"])"
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Consider the following documents, in which the `request_object` is a string representation of a JSON object:
```

Comment:
Do we mean "in which the `request_object` is a JSON object represented as a string"?